pkg/dockerregistry/server/signature.go

```diff
@@ -0,0 +1,83 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+
+	ctxu "github.com/docker/distribution/context"
+
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/registry/handlers"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	imageapiv1 "github.com/openshift/origin/pkg/image/api/v1"
+
+	gorillahandlers "github.com/gorilla/handlers"
+)
+
+func SignatureDispatcher(ctx *handlers.Context, r *http.Request) http.Handler {
+	reference := ctxu.GetStringValue(ctx, "vars.reference")
+
+	signatureHandler := &signatureHandler{
+		Context:   ctx,
+		Reference: reference,
+	}
+
+	return gorillahandlers.MethodHandler{
+		"GET": http.HandlerFunc(signatureHandler.Get),
+	}
+}
+
+type signatureHandler struct {
+	Context   *handlers.Context
+	Reference string
+}
+
+func (s *signatureHandler) Get(w http.ResponseWriter, req *http.Request) {
+	context.GetLogger(s.Context).Debugf("(*signatureHandler).Get")
+	client, ok := UserClientFrom(s.Context)
+	if !ok {
+		context.GetLogger(s.Context).Debugf("(*signatureHandler).Get: unable to get openshift client")
+		return
+	}
+
+	ref, err := imageapi.ParseDockerImageReference(s.Reference)
+	if err != nil {
+		context.GetLogger(s.Context).Debugf("(*signatureHandler).Get: %v", err)
+		return
+	}
+
+	image, err := client.ImageStreamImages(ref.Namespace).Get(ref.Name, ref.ID)
+	if err != nil {
```

Comment:
No `forbidden`. Otherwise, the client will be able to obtain information about the presence of the image on which he has no rights.